spec/cc/engine/violation_spec.rb

```diff
@@ -0,0 +1,48 @@
+require "spec_helper"
+require "cc/engine/duplication"
+
+module CC::Engine::Analyzers
+  RSpec.describe Violation do
+    let(:source) do
+      <<-SOURCE
+        begin
+          foo
+        rescue SyntaxError => e
+          Jekyll.logger.warn "YAML Exception reading \#{File.join(base, name)}: \#{e.message}"
+        rescue Exception => e
+          Jekyll.logger.warn "Error reading file \#{File.join(base, name)}: \#{e.message}"
+        end
+      SOURCE
+    end
+    let(:flay) do
+      Flay.new({
+        diff: false,
+        mass: CC::Engine::Analyzers::Ruby::Main::DEFAULT_MASS_THRESHOLD,
+        summary: false,
+        verbose: false,
+        number: true,
+        timeout: 10,
+        liberal: false,
+        fuzzy: false,
+        only: nil,
+      })
+    end
+
+    describe "violation location" do
+      it "gets appropriate locations for rescue blocks" do
+        sexp = RubyParser.new.process(source, "file.rb")
+        flay.process_sexp(sexp)
+
+        report = flay.analyze[0]
+        violation = Violation.new(10, report, flay.hashes[report.structural_hash])
+
+        location = violation.send(:format_location)
```

Comment:
I'm calling private methods here in the interest of only testing locations, which really is all I care about here. I can shift to calling `format` and inspecting those results if people prefer.